Source/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.m

```diff
@@ -0,0 +1,206 @@
+//
+//  BSG_KSMachHeaders.m
+//  Bugsnag
+//
+//  Created by Robin Macharg on 04/05/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#import <mach-o/dyld.h>
+#import <dlfcn.h>
+#import <Foundation/Foundation.h>
+#import "BSG_KSDynamicLinker.h"
+#import "BSG_KSMachHeaders.h"
+
+BSG_Mach_Binary_Images *bsg_get_mach_binary_images() {
+    return &bsg_mach_binary_images;
+}
+
+// MARK: - Code synchronisation wrappers
+
+// os_unfair_lock is available from specific OS versions onwards:
+//     https://developer.apple.com/documentation/os/os_unfair_lock
+//
+// It deprecates OSSpinLock:
+//     https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/spinlock.3.html
+
+void lock() {
+#if defined(__IPHONE_10_0) || defined(__MAC_10_12) || defined(__TVOS_10_0) || defined(__WATCHOS_3_0)
+    os_unfair_lock_lock(&bsg_mach_binary_images_access_lock);
+#else
+    OSSpinLockLock(&bsg_mach_binary_images_access_lock);
+#endif
+}
+
+void unlock() {
+#if defined(__IPHONE_10_0) || defined(__MAC_10_12) || defined(__TVOS_10_0) || defined(__WATCHOS_3_0)
+    os_unfair_lock_unlock(&bsg_mach_binary_images_access_lock);
+#else
+    OSSpinLockUnlock(&bsg_mach_binary_images_access_lock);
+#endif
+}
```

Comment:
Method names need to be expanded with `bsg_` prefix and probably more description.

Is it worth making a define for the method name to use for lock and unlock in the header and not having separate functions and repeated conditionals in this file?

Reply:
Thanks for the comment - that does indeed lead to a neater implementation.